kotlinx-coroutines-core/wasmJs/src/JSDispatcher.kt

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2016-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines
+
+import kotlinx.coroutines.internal.*
+import org.w3c.dom.*
+import kotlin.coroutines.*
+import kotlin.js.*
+
+private const val MAX_DELAY = Int.MAX_VALUE.toLong()
+
+private fun delayToInt(timeMillis: Long): Int =
```

Comment:
According to IDEA-generated diff, this file is almost identical to the other `JSDispatcher`.

It has a few differences, that can be ironed out:

* A single `clearTimeout` function is not available on d8
* some workarounds that can be unified (e.g. `internal val processQueue: () -> Unit = ::process` vs `internal val processQueue: dynamic = { process() }`
* `external setTimeout` vs `js("window.setTimeout(handler, timeout)")`

Please avoid copy-pasting parts that do not have to be copy-pasted, it will be really hard to maintain it later

Reply:
Done